pkg/deployment/informers.go

```diff
@@ -143,7 +143,10 @@ func (d *Deployment) listenForSecretEvents(stopCh <-chan struct{}) {
 				}
 			},
 			UpdateFunc: func(oldObj, newObj interface{}) {
-				if _, ok := getSecret(newObj); ok {
```

Comment:
While this does work, it is quite unexpected here. 

To keep the design of the operator consistent, this logic should be placed in the inspection loop of the deployment, which is located at `pkg/deployment/deployment_inspector.go`.

Reply:
changed